TACC/Core-Styles
# feat: do NOT truncate buttons by default

## Overview

1. Do **not** truncate buttons by default.
2. Add new class `--can-truncate` to truncate buttons.
3. [I asked Designers (internal communication). I got third approval.](https://tacc-team.slack.com/archives/C060NU133D5/p1700674031041149)

<details><summary><strong>Why?</strong></summary>

- I see CMS and Portals turn it off.
- I see it make bad UI less bad, but not good.
- [Designers C.J. and M.S. voted no to it (internal comm.).](https://tacc-team.slack.com/archives/G01K9ET7JM7/p1680031829746819)
- [Designer H.P. turned it off for expected use case.](https://github.com/TACC/Core-Styles/assets/62723358/c67f1446-46d6-4bdd-8998-4d12ed16b6e2)
- Designer M.S.'s alternative to text wrap is reword when narrow (not truncation).

| anti-use | screenshot |
| - | - |
| client removed truncation 1 | <img width="1200" alt="client removed truncation 1" src="https://github.com/TACC/Core-Styles/assets/62723358/f4d87e18-2bdd-4993-b63f-23b8eaa93663"> |
| client removed truncation 2 | <img width="1200" alt="client removed truncation 2" src="https://github.com/TACC/Core-Styles/assets/62723358/c67f1446-46d6-4bdd-8998-4d12ed16b6e2"> |
| client should remove truncation | <img width="1200" alt="client should remove truncation" src="https://github.com/TACC/Core-Styles/assets/62723358/de27975a-b48a-4f2a-a428-783d7fb3e10b"> |
| only makes layout bug less bad | <img width="1200" alt="truncation only makes bug less bad" src="https://github.com/TACC/Core-Styles/assets/62723358/775c4ffa-9dfd-4a58-88b7-ce345610cfc2"> |

</details>

> [!WARNING]
> Known **side effect** is that [an unresponsive layout will look slightly worse](https://github.com/TACC/Core-Styles/assets/62723358/1839491f-e328-4d06-a455-4a63582ca259). There may be others.

## Related

- replaces #148

## Changes

- **removed** truncation from default button
- **added** `c-button--can-truncate` and `button--truncate` classes
- **added** "Can Truncate" demo pattern variation

## Testing

1. Load http://localhost:3000/components/detail/c-button--secondary.
2. Verify **no** button is truncated.
3. Load http://localhost:3000/components/detail/c-button--can-truncate.
4. Verify "Width" buttons **are** truncated.

## UI

| secondary — _default in Portal_ | truncate |
| - | - |
| ![default](https://github.com/TACC/Core-Styles/assets/62723358/e280edcc-1c75-4963-848e-1704ba103dc0) | ![--can-truncate](https://github.com/TACC/Core-Styles/assets/62723358/15757461-b224-4bb1-a36f-9ee4614b6a98) |
